File: ClientApp/src/app/clases/curso.ts
export class curso{
    id : number;
    nombre : string;
    descripcion : string;
    nivel : number;
    idcarrera : number;  

    constructor(obj){
        //el orden de os parametros afecta la vista de los input en el modal
        this.nombre = obj.nombre;
        this.nivel =  +obj.nivel;
        this.idcarrera = +obj.idcarrera;
        this.descripcion = obj.descripcion;
        this.id = +obj.id; // formateo a number el string // parseInt(val
    }
}

File: modelo/curso.cs
using System.Data;
using System;
    public class curso : oObjeto
    {
         public string Nombre { get; set; }
        public string Descripcion { get; set; }
        public int Nivel { get; set; }
        public int Idcarrera { get; set; }
        
        public curso(){}
        public curso(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Nombre = dr["nombre"].ToString();
        this.Descripcion = dr["descripcion"].ToString();
        this.Nivel = Convert.ToInt32(dr["nivel"]);
        this.Idcarrera = Convert.ToInt32(dr["idcarrera"]);
    }

    }
 